sm_common/include/sm/assert_macros.hpp

```diff
@@ -232,6 +232,21 @@ namespace sm {
       sm::detail::sm_throw_exception<exceptionType>("[" #exceptionType "] ", __FUNCTION__,__FILE__,__LINE__,sm_assert_stringstream.str()); \
     }
 
+#define SM_ASSERT_FINITE(exceptionType, value, message) \
+  if(!(isfinite((value))))           \
```

Comment:
Ah, okay. Yes, then you are right. Probably one should switch at some point. But find for now.